force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -171,17 +272,34 @@ public with sharing class FlowControlLoopsCollections {
         }
         */
 
+        String myOtherGreeting = ''; 
+
+        for (Integer i = 0; i < repeatCount; i++) {
+            myOtherGreeting += inputString;   
+            if (i < repeatCount - 1) {
+              myOtherGreeting += '; ';  
+            }   
+           
+        }  
+
```

Comment:
Check out the String methods repeat and removeEnd().  

https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_string.htm